setup.py

```diff
@@ -53,38 +45,64 @@ def get_version():
                                     stdout=subprocess.PIPE, stderr=subprocess.PIPE)
             rev, err = proc.communicate()
             if proc.returncode == 0:
-                revision = rev.strip()[:8].decode('ascii')
+                revision = rev.strip().decode('ascii')
         except OSError:
             pass
+    return revision
+
 
+def write_version_file(filename, version, revision):
     # Write revision file (only if it needs to be changed)
-    content = 'revision = "{0}"\n'.format(revision)
+    content = '''
+__version__ = "{0}"
+__githash__ = "{1}"
+__release__ = {2}
+    '''.format(version, revision, 'dev' in version)
+
     old_content = None
-    fn = os.path.join(basedir, 'asv', '_version.py')
-    if os.path.isfile(fn):
-        with open(fn, 'r') as f:
+    if os.path.isfile(filename):
+        with open(filename, 'r') as f:
             old_content = f.read()
     if content != old_content:
-        with open(fn, 'w') as f:
+        with open(filename, 'w') as f:
             f.write(content)
 
-    # The revision is not added to the version given to setup(), since
-    # that disturbs 'setup.py develop'
-    return version
+
+version = '0.1dev'
```

Comment:
`rev-list --count` does not tell the user which git version they are running. Better (and PEP440) compatible option would be to either use the form `dev0+HASH` or `devN+HASH`, where `N` is the rev-list count, and `HASH` the git hash.